src/pkg/repository/repository.go

```diff
@@ -288,3 +302,45 @@ func (r repository) DeleteBackup(ctx context.Context, id model.StableID) error {
 
 	return sw.DeleteBackup(ctx, id)
 }
+
+// ---------------------------------------------------------------------------
+// Repository ID Model
+// ---------------------------------------------------------------------------
+
+// repositoryModel identifies the current repository
+type repositoryModel struct {
+	model.BaseModel
+}
+
+const (
+	repositoryID = "repository-id"
+	repoIDTag    = "repo-id"
+)
+
+// should only be called on init.
+func newRepoModel(repoID string) repositoryModel {
+	return repositoryModel{
+		BaseModel: model.BaseModel{
+			ID: repositoryID,
+			Tags: map[string]string{
+				repoIDTag: repoID,
+			},
+		},
+	}
+}
+
+// retrieves the repository info
+func getRepoModel(ctx context.Context, ms *kopia.ModelStore) (*repositoryModel, error) {
+	rm := &repositoryModel{}
+	return rm, ms.Get(ctx, model.RepositorySchema, repositoryID, rm)
+}
+
+// should only be called following a successful init.
+func (rm repositoryModel) write(ctx context.Context, ms *kopia.ModelStore) error {
+	return ms.Put(ctx, model.RepositorySchema, &rm)
+}
```

Comment:
this should be inlined instead of as a separate function. With it as a function on the model it becomes confusing because it's not immediately clear where it's being written to nor does it follow the pattern for how other models are stored